pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_synthetic_privileges failed

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_synthetic_privileges [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/17523562?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/17523562?buildTab=artifacts#/) on release-24.2 @ [b22c6c1d3532ebbbe2c05d3e8699572a864a85e2](https://github.com/cockroachdb/cockroach/commits/b22c6c1d3532ebbbe2c05d3e8699572a864a85e2):


```
=== RUN   TestTenantLogic_synthetic_privileges
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/27b0f33bb7cb86e641d58b0f0758e860/logTestTenantLogic_synthetic_privileges3370247485
    test_log_scope.go:76: use -show-logs to present logs inline
[05:41:34] setting distsql_workmem='57942B';
[05:41:34] setting distsql_workmem='57942B';
[05:41:37] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3308/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 60 statements
    logic.go:3025: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3308/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:306: SELECT username, path, privileges, grant_options FROM system.privileges ORDER BY 1,2
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_ABORT_SPAN): "sql txn" meta={id=b5f49227 key=/Tenant/10/Table/52/1/"testuser4"/"/vtable/crdb_internal/feature_usage" iso=Serializable pri=100.00000000 epo=0 ts=1730353297.505842348,0 min=1730353297.505842348,0 seq=4} lock=true stat=ABORTED rts=1730353297.505842348,0 wto=false gul=1730353298.005842348,0
[05:41:43] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3308/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 69 statements
[05:41:43] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3308/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges with config 3node-tenant: 69 tests, 1 failures
    logic.go:4200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3308/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:319: error while processing
    logic.go:4200: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3308/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:319: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/27b0f33bb7cb86e641d58b0f0758e860/logTestTenantLogic_synthetic_privileges3370247485
--- FAIL: TestTenantLogic_synthetic_privileges (10.83s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_synthetic_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-43819

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_synthetic_privileges [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17644949?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17644949?buildTab=artifacts#/) on release-24.2 @ [16ae1b0bcbdfdf79e7432af0c30e74120807d6eb](https://github.com/cockroachdb/cockroach/commits/16ae1b0bcbdfdf79e7432af0c30e74120807d6eb):


```
=== RUN   TestTenantLogic_synthetic_privileges
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a87a4bfc23f47f52891e3d48f96a6423/logTestTenantLogic_synthetic_privileges2623465742
    test_log_scope.go:76: use -show-logs to present logs inline
[20:17:07] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16037/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 60 statements
[20:17:14] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16037/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 68 statements
    logic.go:3027: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16037/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:306: SELECT username, path, privileges, grant_options FROM system.privileges ORDER BY 1,2
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=d71d2f9b key=/Tenant/10/Table/52/1/"testuser4"/"/vtable/crdb_internal/feature_usage" iso=Serializable pri=100.00000000 epo=0 ts=1731010628.073045618,0 min=1731010628.073045618,0 seq=4} lock=true stat=PENDING rts=1731010628.073045618,0 wto=false gul=1731010628.573045618,0
[20:17:14] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16037/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges with config 3node-tenant: 69 tests, 1 failures
    logic.go:4202: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16037/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:319: error while processing
    logic.go:4202: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16037/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:319: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a87a4bfc23f47f52891e3d48f96a6423/logTestTenantLogic_synthetic_privileges2623465742
--- FAIL: TestTenantLogic_synthetic_privileges (10.29s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_synthetic_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_synthetic_privileges [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17670606?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17670606?buildTab=artifacts#/) on release-24.2 @ [5764366875adca70c3a3ff3d4d13aaf599bc53a7](https://github.com/cockroachdb/cockroach/commits/5764366875adca70c3a3ff3d4d13aaf599bc53a7):


```
=== RUN   TestTenantLogic_synthetic_privileges
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a87a4bfc23f47f52891e3d48f96a6423/logTestTenantLogic_synthetic_privileges1709135763
    test_log_scope.go:76: use -show-logs to present logs inline
[00:00:13] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16007/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 59 statements
    logic.go:3027: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16007/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:306: SELECT username, path, privileges, grant_options FROM system.privileges ORDER BY 1,2
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_ABORT_SPAN): "sql txn" meta={id=3aac9627 key=/Tenant/10/Table/52/1/"testuser4"/"/vtable/crdb_internal/feature_usage" iso=Serializable pri=100.00000000 epo=0 ts=1731110413.554223870,0 min=1731110413.554223870,0 seq=4} lock=true stat=ABORTED rts=1731110413.554223870,0 wto=false gul=1731110414.054223870,0
[00:00:19] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16007/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 69 statements
[00:00:19] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16007/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges with config 3node-tenant: 69 tests, 1 failures
    logic.go:4202: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16007/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:319: error while processing
    logic.go:4202: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16007/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:319: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a87a4bfc23f47f52891e3d48f96a6423/logTestTenantLogic_synthetic_privileges1709135763
--- FAIL: TestTenantLogic_synthetic_privileges (10.36s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_synthetic_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_synthetic_privileges [failed](https://tanzanite.cluster.engflow.com/invocations/default/fd09ea3a-00c8-4ede-858c-6c9a2a315888?testReportRun=12&testReportShard=35&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on release-24.2 @ [287e165b88ff5aa1aa3c9b9ff1303d859c1f960e](https://github.com/cockroachdb/cockroach/commits/287e165b88ff5aa1aa3c9b9ff1303d859c1f960e):


```
=== RUN   TestTenantLogic_synthetic_privileges
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestTenantLogic_synthetic_privileges2857822509
    test_log_scope.go:76: use -show-logs to present logs inline
[13:20:04] setting distsql_workmem='86479B';
[13:20:04] setting distsql_workmem='86479B';
[13:20:06] --- progress: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 51 statements
    logic.go:3027: 
         
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:320: SELECT has_table_privilege('testuser4', 'crdb_internal.feature_usage', 'SELECT')
        expected success, but found
        (40001) restart transaction: has_table_privilege(): TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_ABORT_SPAN): "sql txn" meta={id=b54f32c2 key=/Tenant/10/Table/52/1/"testuser4"/"/vtable/crdb_internal/feature_usage" iso=Serializable pri=100.00000000 epo=0 ts=1731590407.792328367,1 min=1731590407.786104506,0 seq=4} lock=true stat=ABORTED rts=1731590407.786104506,0 wto=false gul=1731590408.286104506,0
        eval.go:2148: in MaybeWrapError()
[13:20:12] --- progress: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 70 statements
[13:20:12] --- done: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges with config 3node-tenant: 70 tests, 1 failures
    logic.go:4202: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:325: error while processing
    logic.go:4202: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:325: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogic_synthetic_privileges2857822509
--- FAIL: TestTenantLogic_synthetic_privileges (10.66s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=12</code>
 - <code>shard=35</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #134806 pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_synthetic_privileges failed [C-test-failure O-robot T-sql-queries branch-release-24.2.5-rc release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_synthetic_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/logictest/testdata/logic_test/synthetic_privileges
+++ b/pkg/sql/logictest/testdata/logic_test/synthetic_privileges
@@ -303,19 +303,6 @@ BEGIN TRANSACTION PRIORITY HIGH ISOLATION LEVEL SERIALIZABLE;
 statement ok
 GRANT SELECT ON crdb_internal.feature_usage TO testuser4
 
-query TTTT
-SELECT username, path, privileges, grant_options FROM system.privileges ORDER BY 1,2
-----
-public     /vtable/crdb_internal/feature_usage  {}                                         {}
-root       /global/                             {MODIFYCLUSTERSETTING}                     {}
-testuser   /externalconn/foo                    {USAGE}                                    {}
-testuser   /global/                             {MODIFYCLUSTERSETTING}                     {}
-testuser   /vtable/crdb_internal/tables         {SELECT}                                   {}
-testuser2  /externalconn/foo                    {USAGE}                                    {}
-testuser2  /global/                             {MODIFYCLUSTERSETTING}                     {}
-testuser3  /global/                             {EXTERNALCONNECTION,MODIFYCLUSTERSETTING}  {}
-testuser4  /vtable/crdb_internal/feature_usage  {SELECT}                                   {}
-
 # This should not cache the uncommitted privilege.
 query B
 SELECT has_table_privilege('testuser4', 'crdb_internal.feature_usage', 'SELECT')
@@ -324,18 +311,6 @@ true
 
 statement ok
 ROLLBACK
-
-query TTTT
-SELECT username, path, privileges, grant_options FROM system.privileges ORDER BY 1,2
-----
-public     /vtable/crdb_internal/feature_usage  {}                                         {}
-root       /global/                             {MODIFYCLUSTERSETTING}                     {}
-testuser   /externalconn/foo                    {USAGE}                                    {}
-testuser   /global/                             {MODIFYCLUSTERSETTING}                     {}
-testuser   /vtable/crdb_internal/tables         {SELECT}                                   {}
-testuser2  /externalconn/foo                    {USAGE}                                    {}
-testuser2  /global/                             {MODIFYCLUSTERSETTING}                     {}
-testuser3  /global/                             {EXTERNALCONNECTION,MODIFYCLUSTERSETTING}  {}
 
 query B
 SELECT has_table_privilege('testuser4', 'crdb_internal.feature_usage', 'SELECT')

